jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -97,9 +97,19 @@ public ImagesAndRegistryClient call()
     // Skip this step if this is a scratch image
     ImageReference imageReference = buildContext.getBaseImageConfiguration().getImage();
     if (imageReference.isScratch()) {
+      String architecture = "";
+      String os = "";
+      for (Platform platform : buildContext.getContainerConfiguration().getPlatforms()) {
```

Comment:
If multiple platforms are given, this method should return multiple `Image`s (i.e., `List<Image>` as `ImagesAndRegistryClient.images`). That is, it shouldn't use `Collections.singletonList()` on line 112.

We should also add a test for the case of multiple platforms.

Reply:
Right, good call. Changed this method to add multiple platforms.